Tests/UnitTests/Core/Vertexing/AdaptiveGridTrackDensityTests.cpp

```diff
@@ -0,0 +1,341 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/tools/output_test_stream.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Surfaces/PerigeeSurface.hpp"
+#include "Acts/Tests/CommonHelpers/FloatComparisons.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Units.hpp"
+#include "Acts/Vertexing/AdaptiveGridTrackDensity.hpp"
+
+namespace bdata = boost::unit_test::data;
+using namespace Acts::UnitLiterals;
+
+namespace Acts {
+namespace Test {
+
+using Covariance = BoundSymMatrix;
+
+BOOST_AUTO_TEST_CASE(adaptive_gaussian_grid_density_track_adding_test) {
+  const int trkGridSize = 15;
+
+  double binSize = 0.1;  // mm
+
+  // Set up grid density with zMinMax
+  AdaptiveGridTrackDensity<trkGridSize>::Config cfg(binSize);
+  AdaptiveGridTrackDensity<trkGridSize> grid(cfg);
+
+  // Create some test tracks
+  Covariance covMat;
+  covMat << 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0,
```

Comment:
can't you just set this to unity?

Reply:
if not, please consider breaking the lines so it can be read as a matrix

Reply:
Done. Thanks.